code/layout-analyzer.js

```diff
@@ -68,6 +70,49 @@ window.addEventListener('DOMContentLoaded', () => {
     element.querySelector(sel).innerText = fmtPercent(num, precision);
   };
 
+  const showPercentAll = (sel, nums, precision, parentId) => {
+    const element = parentId
+      ? document.querySelector(parentId).shadowRoot
+      : document;
+    element.querySelector(sel).innerText =
+      nums.map(value => fmtPercent(value, precision)).join(' / ');
+  };
+
+  const sumUpBar = bar => bar.good + bar.meh + bar.bad;
+  const sumUpBarGroup = group => group.reduce((acc, bar) => acc + sumUpBar(bar), 0);
+
+  const getKeyPositionQuality = keyCode => {
+    // This is cursed, there *has* to be a better way
```

Comment:
Agreed. Defining two key sets (`good`, `meh`) and checking whether `keyCode` belongs to one of them would be smarter.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
I almost never use sets, completely forgot they even existed lol